File: mobile/lib/models/customer.dart
class Customer {
  final String id;
  final String name;
  final String phone;
  final String? email;
  final String? address;
  final String? kycId;
  final String kycType;
  final DateTime joinedDate;
  final bool isActive;
  final String? notes;

  Customer({
    required this.id,
    required this.name,
    required this.phone,
    this.email,
    this.address,
    this.kycId,
    required this.kycType,
    required this.joinedDate,
    required this.isActive,
    this.notes,
  });

  factory Customer.fromJson(Map<String, dynamic> json) {
    return Customer(
      id: json['_id'],
      name: json['name'],
      phone: json['phone'],
      email: json['email'],
      address: json['address'],
      kycId: json['kyc_id'],
      kycType: json['kyc_type'] ?? 'aadhaar',
      joinedDate: DateTime.parse(json['joined_date']),
      isActive: json['isActive'] ?? true,
      notes: json['notes'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'name': name,
      'phone': phone,
      'email': email,
      'address': address,
      'kyc_id': kycId,
      'kyc_type': kycType,
      'notes': notes,
      'isActive': isActive,
    };
  }
}

File: backend/models/Customer.js
const mongoose = require('mongoose');

const customerSchema = new mongoose.Schema({
  name: {
    type: String,
    required: [true, 'Please provide customer name'],
    trim: true
  },
  phone: {
    type: String,
    required: [true, 'Please provide phone number'],
    trim: true
  },
  email: {
    type: String,
    trim: true,
    lowercase: true
  },
  address: {
    type: String,
    trim: true
  },
  kyc_id: {
    type: String,
    trim: true
  },
  kyc_type: {
    type: String,
    enum: ['aadhaar', 'pan', 'voter_id', 'driving_license', 'other'],
    default: 'aadhaar'
  },
  joined_date: {
    type: Date,
    default: Date.now
  },
  isActive: {
    type: Boolean,
    default: true
  },
  notes: {
    type: String,
    trim: true
  },
  createdBy: {
    type: mongoose.Schema.Types.ObjectId,
    ref: 'Employee',
    required: true
  }
}, {
  timestamps: true
});

module.exports = mongoose.model('Customer', customerSchema);
